src/google/cloud/ndb/key.py

```diff
@@ -852,6 +858,16 @@ def get():
                 except KeyError:
                     pass
 
+            remote_cache_locked = False
+            if use_remote_cache:
+                result = yield remote_cache.cache_get(self)
+                remote_cache_locked = remote_cache.is_locked_value(result)
+                if result is not None and not remote_cache_locked:
+                    return result
+                if result is None:
+                    yield remote_cache.cache_set_locked(self)
+                    yield remote_cache.cache_start_cas(self)
```

Comment:
I'm noticing here that when ``remote_cache_locked`` is ``True``, we call ``cache_start_cas`` but then, below, we don't call ``cache_cas``. If I'm reading this right, there's not really any reason to call ``cache_start_cas`` if it's locked, because we're not then going to try and call ``cache_cas``. Let me know if I'm reading this incorrectly.

Reply:
```remote_cache_locked``` actually means the result value is 0, so in this case remote_cache_locked should be ```False```.
* if result == Entity,  cache is set, so return it.
* if result == 0 (or b'0' or '0'), cache is set and locked, so do nothing for cache.
* if result == None, cache is not set (so not locked), so start check and set session.

And this lock with 0 mechanism is quite fragile, because put() or delete() also set 0 value.
If any updating action started between cache_set_locked and cache_set, setting cache from get() should fail. So cache_start_cas and cache_cas do the work.

This came from original implementation in py27, and sounds reasonable for me.